pay_ios/ios/Classes/UtilityExtensions.swift

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PassKit
+
+/// A set of utility methods associated to `PKPayment`.
+extension PKPayment {
+
+  /// Creates a `Dictionary` representation of the `PKPayment` object.
+  public func toDictionary() -> [String: Any] {
+    return [
+      "token": String(decoding: token.paymentData, as: UTF8.self) as Any?,
+      "paymentMethod": token.paymentMethod.toDictionary(),
+      "billingContact": billingContact?.toDictionary(),
+      "shippingContact": shippingContact?.toDictionary(),
+      "shippingMethod": shippingMethod?.toDictionary(),
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PKPaymentMethod`.
+extension PKPaymentMethod {
+
+  /// Creates a `Dictionary` representation of the `PKPaymentMethod` object.
+  public func toDictionary() -> [String: Any] {
+    return [
+      "displayName": displayName as Any?,
+      "network": network?.rawValue,
+      "type": type.rawValue
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PKContact`.
+extension PKContact {
+
+  /// Creates a `Dictionary` representation of the `PKContact` object.
+  public func toDictionary() -> [String: Any] {
+    return [
+      "name": name?.toDictionary() as Any?,
+      "emailAddress": emailAddress,
+      "phoneNumber": phoneNumber?.stringValue,
+      "postalAddress": postalAddress?.toDictionary()
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PersonNameComponents`.
+extension PersonNameComponents {
+
+  /// Creates a `Dictionary` representation of the `PersonNameComponents` object.
+  public func toDictionary() -> [String: Any]? {
+    let dict = [
+      "namePrefix": (namePrefix != nil && !namePrefix!.isEmpty) ? namePrefix : nil as Any?,
```

Comment:
In line with the no-post-process approach we are taken with the name, I think it'd be useful to leave these elements untouched and send them back the same way as Apple Pay returns them. Not only this makes it easier to troubleshoot a potentially flawed Apple Pay integration but it also keeps the thin-layer principle consistent.

Reply:
Just to confirm, you mean leaving the empty string value `""` as is, right?

Reply:
That's right, or to any value that the Apple Pay lib gives it, and keep it true to that.